pkg/ebpf/events_derived.go

```diff
@@ -81,11 +53,19 @@ func (t *Tracee) deriveEvents(ctx context.Context, in <-chan *trace.Event) (<-ch
 				out <- event
 
 				// Derive event before parsing its arguments
-				derivatives := t.deriveEvent(*event)
+				derivatives, errors := events.Derive(*event, t.eventDerivations)
 
 				for _, derivative := range derivatives {
+					// Drop derived events which were not requested by the user
+					if !t.events[events.ID(derivative.EventID)].emit {
+						continue
+					}
```

Comment:
Hmmm, I think a better option would be to have an `enabled` flag in the deriving table. Then we could use the .emit flag to determine it. WDYT?
I'm trying to avoid what you suggested because i've tried to reduce the coupling here, the benefit was adding tests to the mechanism (and I think it makes the code nicer :))

Reply:
Yes, having a flag is a better option, although I'm not sure `enabled` is the right name (as an event can be submitted to userspace, but not emitted as an output)

Reply:
But derived events are always after the submit stage and are created in user space, so only emit is relevant to them.

Reply:
Yep. Your'e right on that. Let's go with the enabled flag